### PR TITLE
Tangent function, DEBUG.md documentation, corrected version of strncpy() and variable unification were added.

### PR DESCRIPTION
**Pull request by angelpilled**
[43c7896](https://github.com/lvntky/ArtilleryOS/pull/51/commits/43c7896818bc241ea72bad8a7223bec9622fd2b8) [453bd65](https://github.com/lvntky/ArtilleryOS/pull/51/commits/453bd65766d9a44bcb5ab609ae9631fd202b74af):
I added a tangent function in the math library and DEBUG.md documentation of GDB debugging was made with the help of @moussacodes :)

[6987833](https://github.com/lvntky/ArtilleryOS/pull/51/commits/6987833b48dc8fb44c8216997a5190ef6d90d9e4):
I further corrected the strncpy() function inside string.c so it checks **source** and NOT **destination** in the while condition. This makes it so if we type in more from `n` characters than enough `source` characters are there, it fills up the remaining space with null terminators.

[9f02ce4](https://github.com/lvntky/ArtilleryOS/pull/51/commits/9f02ce445e272b46e4409741b7b0a52723f7717e):
DEBUG.md formatting was fixed & beautified!

### CODE REVIEW

File: docs/DEBUG.md
Comment (lvntky) [inline docs/DEBUG.md:1]: nice one, tank you!

**Review by lvntky — APPROVED** [review]
more understandable. thank you

**Review by lvntky — APPROVED** [review]
good commit!

File: docs/DEBUG.md
Comment (angelpilled) [reply docs/DEBUG.md:1]: You're welcome!

**Pull request merged**